### PR TITLE
openthread: update certification documentation

### PR DESCRIPTION
**Pull request by edmont**
This commit clarifies which OpenThread libraries have Thread Certification granted for each device and NCS release.

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>

KRKNWK-8126

NCS PR: https://github.com/nrfconnect/sdk-nrf/pull/3396

### CODE REVIEW

**Review by lmaciejonczyk — APPROVED** [review]

**Review by MarekPorwisz — APPROVED** [review]

File: openthread/README.rst [hunk 1]
```diff
@@ -16,182 +16,36 @@ For more information about OpenThread feature sets and how to select them in |NC
...
+     - nRF52833
+     - nrf_oberon
+     - 1.1
+     - ``07f430dac6c400abdd358ba1173c0f94b94a5a80``
+     - ``9-2019-q4-major``
+     - :option:`CONFIG_OPENTHREAD_NORDIC_LIBRARY_MASTER`
+
+For a comprehensive list of the used Kconfig options check the file :file:`openthread_lib_configuration.txt` within each library folder.
+
+.. note:: Some of the available libraries for each NCS release may have not been certified.
```
Comment (pszkotak) [inline openthread/README.rst:51]: This sentence rises a question how many more are there.
I'd rephrase a bit.
```suggestion
.. note:: Only variants shown in the table have been certified.
```

Comment (edmont) [reply openthread/README.rst]: Changed, thanks.

**Review by pszkotak — APPROVED** [review]

File: openthread/README.rst [hunk 2]
```diff
@@ -16,182 +16,36 @@ For more information about OpenThread feature sets and how to select them in |NC
...
 
-Pre-built library variants
-**************************
-
-The OpenThread pre-built libraries were created based on commit ``07f430dac6c400abdd358ba1173c0f94b94a5a80`` of OpenThread,
-using the GNU Arm Embedded Toolchain version ``9-2019-q4-major``.
-
-The following table describes the Kconfig options that were set for the build for each variant.
-For a more concise list of features enabled for each feature set, see :ref:`nrf:thread_ug_feature_sets`.
+The following table describes the available Thread Certified libraries for all |NCS| releases.
```
Comment (ru-fu) [inline openthread/README.rst:22]:
```suggestion
The following table describes the available Thread certified libraries for all |NCS| releases.
```

File: openthread/README.rst [hunk 3]
```diff
@@ -16,182 +16,36 @@ For more information about OpenThread feature sets and how to select them in |NC
...
+     - ``07f430dac6c400abdd358ba1173c0f94b94a5a80``
+     - ``9-2019-q4-major``
+     - :option:`CONFIG_OPENTHREAD_NORDIC_LIBRARY_MASTER`
+   * - v1.4.0
+     - nRF52833
+     - nrf_oberon
+     - 1.1
+     - ``07f430dac6c400abdd358ba1173c0f94b94a5a80``
+     - ``9-2019-q4-major``
+     - :option:`CONFIG_OPENTHREAD_NORDIC_LIBRARY_MASTER`
```
Comment (ru-fu) [inline openthread/README.rst:47]:
```suggestion
   * - SoC
     - |NCS| tag
     - Crypto backend
     - Thread version
     - OpenThread commit
     - GNU Arm Embedded Toolchain
     - Feature set
   * - nRF52840
     - v1.4.0
     - CC310
     - v1.1
     - ``07f430dac6c400abdd358ba1173c0f94b94a5a80``
     - ``9-2019-q4-major``
     - :option:`CONFIG_OPENTHREAD_NORDIC_LIBRARY_MASTER`
   * - nRF52833
     - v1.4.0
     - nrf_oberon
     - v1.1
     - ``07f430dac6c400abdd358ba1173c0f94b94a5a80``
     - ``9-2019-q4-major``
     - :option:`CONFIG_OPENTHREAD_NORDIC_LIBRARY_MASTER`
```
The first two columns should be inverted, because users are interested in qualifications for their chip (and will switch to a different NCS version if necessary).
I'm wondering if Crypto backend and Thread version also should be inverted - I tried to locate the file mentioned below, and when navigating through the libs, you select v1.1 before you select oberon/cc310. But I'm not sure which one is more relevant to the user - probably the Thread version, and then they use the backend that is available?

File: openthread/README.rst [hunk 4]
```diff
@@ -16,182 +16,36 @@ For more information about OpenThread feature sets and how to select them in |NC
...
+
+For a comprehensive list of the used Kconfig options check the file :file:`openthread_lib_configuration.txt` within each library folder.
```
Comment (ru-fu) [inline openthread/README.rst:49]:
```suggestion
For a comprehensive list of the used Kconfig options, check the :file:`openthread_lib_configuration.txt` file within each library folder.
```

Comment (ru-fu) [reply openthread/README.rst:49]: Trying to locate the file, I got lost when I had to decide between ftd, master, and mtd. In the old table, it was clear that there are three options. Here, the only mention of master is in the name of the feature set - is that clear enough?

**Review by ru-fu — CHANGES_REQUESTED** [review]

File: openthread/README.rst (hunk 3 above)
Comment (edmont) [reply openthread/README.rst:47]: Agree about SoC as first column. I'll also move the Thread version column before the crypto backend, which is merely informative here.

File: openthread/README.rst (hunk 4 above)
Comment (edmont) [reply openthread/README.rst:49]: The omission of the other variants it's deliberated because if not certified there are of little use anymore. In any case, the linked `thread-ug-feature-sets` should be explanatory.

Comment (ru-fu) [reply openthread/README.rst]: OK, let's leave as is for now. I'm not sure if it's clear enough here (it is explained in more detail in the link), but we can fix later.

**Review by ru-fu — APPROVED** [review]
